crates/fee/src/lib.rs

```diff
@@ -464,56 +475,87 @@ impl<T: Config> Pallet<T> {
         amount.rounded_mul(<ReplaceGriefingCollateral<T>>::get())
     }
 
-    pub fn withdraw_all_vault_rewards(vault_id: &DefaultVaultId<T>) -> DispatchResult {
-        Self::distribute_from_reward_pool(vault_id)?;
+    pub fn compute_vault_rewards(
+        vault_id: &DefaultVaultId<T>,
+        nominator_id: &T::AccountId,
+        currency_id: CurrencyId<T>,
+    ) -> Result<BalanceOf<T>, DispatchError> {
+        frame_support::storage::with_transaction::<_, DispatchError, _>(|| {
```

Comment:
I'm assuming this is safe to use from the RPC interface as long as we always rollback. @sander2 what do you think? I noticed you followed a similar approach in the integration tests. The alternative is to add a `distribute_and_compute_reward` method to the `Rewards` interface.

Reply:
If this is _only_ used in rpcs, then no rollback should be required at all since rpcs shouldn't be able to make storage changes. If this is also called from with extrinsics, then it _is_ necessary, and afaik it should be safe. The question either way is what happens if you try to (temporarily or permanently) modify chain state in rpcs. We'd need to test that it doesn't just panic